packages/shared/src/components/auth/common.tsx

```diff
@@ -46,6 +57,10 @@ export const providerMap: Record<string, Provider> = {
   },
 };
 
+export const getProviderMapClone = (
```

Comment:
Why do we need the clone?

Reply:
This is just so the original object won't get modified, we had areas where some properties had to be modified.